### PR TITLE
feat(dashmate): additional dashd options

### PR DESCRIPTION
**Pull request by pshenmic**
## Issue being fixed or feature implemented

Merge some templating features for dashd from https://github.com/dashpay/dash-network-deploy. This is needed for dashmate integration for dash-network-deploy.

## What was done?

Added template params for:
* logips
* listen
* allowprivatenet
* indexes
* zmqport
* minimumdifficultyblocks
* powtargetspacing

## How Has This Been Tested?

Locally

## Breaking Changes

Added more fields for core in dashmate config.json

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dashmate/templates/core/dash.conf.dot [hunk 1]
```diff
@@ -60,18 +63,24 @@ addnode={{=seed.host}}:{{=seed.port}}{{~}}
 devnet={{=it.core.devnetName}}
 llmqchainlocks=llmq_devnet
 llmqinstantsend=llmq_devnet
+llmqinstantsenddip0024=llmq_devnet_dip0024
```
[the remark this answers is not in the document]
Comment (pshenmic) [reply packages/dashmate/templates/core/dash.conf.dot:71]: Isnt it devnet only?

File: packages/dashmate/configs/migrations.js [hunk 2]
```diff
@@ -398,4 +398,15 @@ module.exports = {
 
     return configFile;
   },
+  '0.24.0-dev.11': (configFile) => {
+    Object.entries(configFile.configs)
+      .forEach(([, config]) => {
+        config.core.logIps = 0;
+        config.core.indexes = config.name === 'mainnet' ? 0 : 1;
+        config.core.minimumDifficultyBlocks = 1000;
```
Comment (shumkov) [inline packages/dashmate/configs/migrations.js:406]: I guess this setting should depend on the network, isn't it?

File: packages/dashmate/configs/system/base.js [hunk 3]
```diff
@@ -66,6 +66,10 @@ module.exports = {
       },
     },
     debug: 0,
+    listen: 1,
```
Comment (shumkov) [inline packages/dashmate/configs/system/base.js:69]: I believe you meant `logIps`

File: packages/dashmate/templates/core/dash.conf.dot [hunk 4]
```diff
@@ -5,6 +5,9 @@ logtimestamps=1
 maxconnections=256
 printtoconsole=1
 debug={{=it.core.debug }}
+{{? it.core.logIps === true}}
```
Comment (shumkov) [inline packages/dashmate/templates/core/dash.conf.dot:8]: You values only 0 and 1

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dashmate/configs/schema/configJsonSchema.js [hunk 5]
```diff
@@ -218,6 +218,14 @@ module.exports = {
           required: ['docker'],
           additionalProperties: false,
         },
+        minimumDifficultyBlocks: {
+          type: 'integer',
+          minimum: 1,
```
Comment (shumkov) [inline packages/dashmate/configs/schema/configJsonSchema.js:223]: 0

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dashmate/configs/migrations.js (hunk 2 above)
Comment (shumkov) [inline packages/dashmate/configs/migrations.js:406]: I would honestly keep it 0 by default

File: packages/dashmate/configs/system/base.js [hunk 6]
```diff
@@ -66,6 +66,10 @@ module.exports = {
       },
     },
     debug: 0,
+    logIps: 0,
+    indexes: true,
+    minimumDifficultyBlocks: 1000,
```
Comment (shumkov) [inline packages/dashmate/configs/system/base.js:71]: I would keep it 0 by default

File: packages/dashmate/configs/system/base.js [hunk 7]
```diff
@@ -66,6 +66,10 @@ module.exports = {
...
+    powTargetSpacing: 150,
```
Comment (shumkov) [inline packages/dashmate/configs/system/base.js:72]: is it 2.5 min?

Comment (pshenmic) [reply packages/dashmate/configs/system/base.js:72]: Yes, in seconds

**Review by shumkov — DISMISSED** [review]
👍

File: packages/dashmate/configs/schema/configJsonSchema.js [hunk 8]
```diff
@@ -150,6 +150,12 @@ module.exports = {
               type: 'string',
               minLength: 1,
             },
+            allowIps: {
+              type: 'array',
+              items: {
+                type: 'string',
```
Comment (shumkov) [inline packages/dashmate/configs/schema/configJsonSchema.js:156]: format: ipv4

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]
👍

**Review by shumkov — APPROVED** [review]

**Pull request merged**